src/olympia/files/utils.py

```diff
@@ -106,6 +106,12 @@ def id_to_path(pk, breadth=1):
 def get_file(fileorpath):
     """Get a file-like object, whether given a FileUpload object or a path."""
     if hasattr(fileorpath, 'path'):  # FileUpload
+        if not fileorpath.path:
```

Comment:
I don't know which is more "correct", but we have a `FileUpload.submitted` `@property` that checks if `self.addon` is set to determine if the `FileUpload` has been "used" to create a version.

Reply:
(we should probably have a single method of determining whether a `FileUpload` has already been used.  Could be a follow-up)

Reply:
mmm, I don't think we're using that property anywhere. It's wrong/misleading, since nowadays we set `addon` all the time. I guess I could replace it with something like `self.pk and self.path` and use it instead here... but ultimately, what we really care about here is that the path exists...

Reply:
:/ looks like it's only uploads created via the addon submission api that don't have `addon` set until they're consumed to create a `Version`